### PR TITLE
[Merged by Bors] - TY-2799 compact engine state box

### PR DESCRIPTION
**Pull request by Robert-Steiner**
Ticket:
- [TY-2799]

I checked the `HiveActiveSearchRepository` and it has a [`clear`](https://github.com/xaynetwork/xayn_discovery_engine/blob/1bd2a34c484d4b37e11729d0f20cf323202ddf67/discovery_engine/lib/src/infrastructure/repository/hive_active_search_repo.dart#L33) method that we call in the code.

[TY-2799]: https://xainag.atlassian.net/browse/TY-2799?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: discovery_engine/lib/src/domain/event_handler.dart [hunk 1]
```diff
@@ -323,7 +323,14 @@ class EventHandler {
     await Future.wait([
       _openDbBox<Document>(documentBox),
       _openDbBox<ActiveDocumentData>(activeDocumentDataBox),
-      _openDbBox<Uint8List>(engineStateBox),
+
+      /// See TY-2799
+      /// Hive usually compact our boxes automatically. However, with the default
```
Comment (janpetschexain) [inline discovery_engine/lib/src/domain/event_handler.dart:328]:
```suggestion
      /// Hive usually compacts our boxes automatically. However, with the default
```

**Review by janpetschexain — APPROVED** [review]
lgtm

**Review by biernacki — APPROVED** [review]
really nice explanation 👍

**Comment by Robert-Steiner** [comment]
bors merge

**Comment by Robert-Steiner** [comment]
bors merge

**Comment by Robert-Steiner** [comment]
bors merge

**Pull request closed without merging**